File: exercise/python语言进阶/数据结构和算法/排序算法.js
//实现一个排序算法
function sort(array, comp = (x, y) => x < y) {
    const arr = [...array]
    for (let i = 0; i <
    arr.length - 1; i++
    ) {
        min_index = i
        for (let j = i + 1; j < arr.length; j++) {
            if (comp(arr[j], arr[min_index])) {
                min_index = j
            }
        }
        [arr[min_index], arr[i]] = [arr[i], arr[min_index]]
    }
    return arr
}

res = sort([12, 1, 2])
console.log(res)

File: src/02_python语言进阶/数据结构和算法/排序算法.py
def select_sort(items, comp=lambda x, y: x < y):
    # 复制items
    items = items[:]
    for i in range(len(items) - 1):
        min_index = i
        for j in range(i + 1, len(items)):
            if comp(items[j], items[min_index]):
                min_index = j
        items[i], items[min_index] = items[min_index], items[i]
    return items

res = select_sort([12, 0, 1, -2, 12, -100])
print(res)